src/Germany/Mews.Fiscalizations.Germany/ModelMapper.cs

```diff
@@ -64,22 +68,52 @@ internal static ResponseResult<Tss> MapTss(Dto.TssResponse tss)
                 initializedUtc: tss.TimeInit.FromUnixTime(),
                 disabledUtc: tss.TimeDisable.FromUnixTime(),
                 certificate: tss.Certificate,
-                certificateSerial: tss.CertificateSerial,
+                serialNumber: tss.SerialNumber,
                 publicKey: tss.PublicKey,
                 signatureCounter: tss.SignatureCounter,
                 signatureAlgorithm: tss.SignatureAlgorithm,
                 transactionCounter: tss.TransactionCounter
             ));
         }
 
+        internal static ResponseResult<CreateTssResult> MapCreateTss(Dto.CreateTssResponse createTssResponse)
+        {
+            return new ResponseResult<CreateTssResult>(successResult: new CreateTssResult(
+                adminPuk: createTssResponse.AdminPuk,
+                tss: new Tss(
+                    id: createTssResponse.Id,
+                    description: createTssResponse.Description,
+                    state: MapTssState(createTssResponse.State),
+                    createdUtc: createTssResponse.TimeCreation.FromUnixTime(),
+                    initializedUtc: createTssResponse.TimeInit.FromUnixTime(),
+                    disabledUtc: createTssResponse.TimeDisable.FromUnixTime(),
+                    certificate: createTssResponse.Certificate,
+                    serialNumber: createTssResponse.SerialNumber,
+                    publicKey: createTssResponse.PublicKey,
+                    signatureCounter: createTssResponse.SignatureCounter,
+                    signatureAlgorithm: createTssResponse.SignatureAlgorithm,
+                    transactionCounter: createTssResponse.TransactionCounter
+                )
+            ));
+        }
+
         private static TssState MapTssState(Dto.TssState state)
         {
             return state.Match(
+                Dto.TssState.CREATED, _ => TssState.Created,
                 Dto.TssState.DISABLED, _ => TssState.Disabled,
                 Dto.TssState.INITIALIZED, _ => TssState.Initialized,
                 Dto.TssState.UNINITIALIZED, _ => TssState.Uninitialized,
                 _ => throw new NotImplementedException($"Tss state: {state} is not implemented.")
             );
         }
+
+        private static ClientState MapClientState(Dto.ClientState state)
+        {
+            return state.Match(
+                Dto.ClientState.REGISTERED, _ => ClientState.Registered,
+                Dto.ClientState.DEREGISTERED, _ => ClientState.Deregistered
```

Comment:
What the hell is deregistered? 
Like Enabled and disabled? 
Or existing and deleted?

Reply:
Based on my understanding, they're more like Enabled and disabled.

Registered means the client can be used
Deregistered means the client cannot be used

The initial state when creating a client is Registered.